test/parallel/test-node-run.js

```diff
@@ -222,4 +222,46 @@ describe('node --run [command]', () => {
     assert.strictEqual(child.stdout, '');
     assert.strictEqual(child.code, 1);
   });
+
+  it('runs script in a custom working directory using --run-dir', async () => {
+    const customDir = fixtures.path('run-script');
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--run-dir', customDir, '--run', `pwd${envSuffix}` ],
+
+      { cwd: fixtures.path('run-script/sub-directory') }
+    );
+
+    assert.strictEqual(child.stdout.trim(), customDir);
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('returns error on non-directory path for --run-dir', async () => {
+    const nonDirPath = fixtures.path('run-script/package.json');
+
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ '--run-dir', nonDirPath, '--run', 'pwd' ],
+      { cwd: fixtures.path('run-script') }
+    );
+
+    assert.match(child.stderr, /not a directory/);
+    assert.strictEqual(child.stdout, '');
+    assert.strictEqual(child.code, 1);
+  });
+
+  it('returns error when --run-dir is used without --run', async () => {
```

Comment:
on line 263 you validate that stderr is "" and there is no error thrown. How are we testing returning error here?

Reply:
```suggestion
  it('--run-dir should be no-op when used without --run', async () => {
```

Reply:
I had set this up to use the wrong command, thank you for the editing suggestion